airflow/providers/amazon/aws/operators/batch.py

```diff
@@ -177,29 +177,26 @@ def submit_job(self, context: Dict):  # pylint: disable=unused-argument
             self.job_id = response["jobId"]
 
             self.log.info("AWS Batch job (%s) started: %s", self.job_id, response)
-
         except Exception as e:
             self.log.error("AWS Batch job (%s) failed submission", self.job_id)
             raise AirflowException(e)
 
     def monitor_job(self, context: Dict):  # pylint: disable=unused-argument
         """
         Monitor an AWS Batch job
+        monitor_job can raise an exception or an AirflowTaskTimeout can be raised if execution_timeout
+        is given while creating the task. These exceptions should be handled in taskinstance.py
+        instead of here like it was previously done
```

Comment:
what was done previously is generally something that is only of interest in the PR -- not in the code.   in this case i'd just document what it does _now_

and in this case, i would not add any documentation

in general it is assumed that code can throw exceptions.  that a task timeout can be raised is true of all operators.  and you don't need to add a warning when you are not catching an exception

Reply:
@potiuk Wanted me to add a comment describing what is going on. @dstandish I can remove the comment if you and @potiuk both agree

Reply:
It's about the only place where the Airflow Task Timeout exception @dstandish in the entire code. I think it's not at all 'common' knowledge what's going on here. But i do agree that refering to "previous" state is not needed. So if that can be removed maybe we have a nice compromise ;).

Reply:
AirflowTaskTimeout is thrown in [taskinstance.py](https://github.com/apache/airflow/blob/master/airflow/models/taskinstance.py#L1309) code for all operators. 
```
try:
    with timeout(task_copy.execution_timeout.total_seconds()):
        result = task_copy.execute(context=context)
except AirflowTaskTimeout:
    task_copy.on_kill()
    raise
```

Reply:
i am a bit confused @potiuk

> an AirflowTaskTimeout can be raised if execution_timeout is given while creating the task

this is true of every operator -- it's something that's part of base operator right?

maybe i miss something.  but i don't see anywhere in this operator (or its inherited methods) that task timeout is raised explicitly.

Reply:
@dstandish It's part of the base operator, but some hooks/operators have also used this exception. The issue I am trying to solve in the PR is the timeout exception raised when the task is running for more than that the defined interval which is handled by taskinstance

Reply:
> this is true of every operator -- execution_timeout is something that's part of base operator right?

Correct. This is like that but THE ONLY core operator using it is this AWS one

> maybe i miss something. but i don't see anywhere in this operator (or its inherited methods) that task timeout is raised explicitly.

We are following the same path here :). And it proves IMHO that some comment is needed here.

I had the same question initially because I did not see it either. The AWS batch operators are built in a very strange way and while the "batch.py" does  not look like it can throw the timeout, it actually does in here:

https://github.com/apache/airflow/blob/fc67521f31a0c9a74dadda8d5f0ac32c07be218d/airflow/providers/amazon/aws/hooks/datasync.py#L316

Which is several stack frames below (and only if you try hard you can find it out).

And somebody else did not actually know it and added the original try/except Exception (and threw Airflow Execption()) hiding the fact that there is a timeout exception raised deep in the datasync hook. 

So I asked @ayushchauhan0811 to explain it here - otherwise we risk that someone will again unnowingly swallow the Timeout exception.

Reply:
ah yes makes sense.  sorry to consume so much of your time.  and thanks for explaining.

when i did "find in files" i saw this datasync reference but assumed datasync would not be involved in batch hook 🤔.

in any case i am glad to see this change.  we should never catch Exception and reraise with AirflowException (i.e. catch and reraise for no reason) and this is a good example why... i see it somewhat frequently... it only discards information and adds indent.

on the related question of task timeout exception...  curious... i wonder if this exception should be reserved only for an `execution_timeout` timeout, and thou shalt not use it otherwise?  seems like it to me...  i don't think there would be any different outcome for the task if it was `raise Exception('i give up')` -- i.e. there's no special handling behavior e.g. like there is with skip and reschedule is there?

Reply:
Yeah. I also do not like how it is done in AWS. The "with timeout" context manager seems universal solution that should also work for datasync operator, so I am really not sure if it is needed at all :) .

Reply:
@potiuk @dstandish so what should I do? Keep the comment or remove it?

Reply:
keep it! jarek is the boss :)

Reply:
@potiuk if you think this looks good, can we merge it?